app/frontend/package.json

```
@@ -26,7 +26,8 @@
     "format": "prettier --write src/",
     "test-travis": "NODE_ENV=test && react-scripts test --coverage --watchAll=false",
     "storybook": "start-storybook -p 6006 -s public --no-dll",
-    "build-storybook": "build-storybook -s public --no-dll"
+    "build-storybook": "build-storybook -s public --no-dll",
+    "jest": "jest"
```

Shouldn't we use the following to prevent running too many tests?
```suggestion
    "jest": "jest --findRelatedTests"
```

---

I agree we can config jest for smaller test cases. However, I can only see one test for front-end. I think will be better to have more tests, group those tests and then re-config jest.
What do you suggest?

---

more tests I agree with.
Do you mean grouping tests and reconfiguring tests as opposed to using `--findRelatedTests`?

---

Yes

---

As other dev members are against using the pre commit hooks you can make it that way using only a package script and deleting the Husky lines. However, there is already a test script. That means this PR is not needed. But wait what Lucas has to say.

---

Maybe if Aapeli thinks we shouldn't have global pre-commkt hooks, let's take husky out.

This got a bit messed up because of the merge into the main repo, sorry about that! But we can merge the other changes without husky.

---

There is an option to commit without running husky by adding `--non-verify`
But if you guys prefer not to have husky at all, I will remove it

---

Using `--no-verify` basically defeats the point of commit hooks as you're ignoring them at that point. Why have them there if we can't run it?

Is it that bad though if we configure it to run on the `frontend` repo only? `lint-staged` has pretty powerful glob matching patterns that only apply the hook to matching files so won't really cause any issues for the Python backend.

---

I think Aapeli's complaint is not about the hooks being run, but about them being installed (in .git/hooks).

---

So when a commit is done, the hook script in .git/hooks is run, which runs the script in package.json in the husky.hooks property. Which typically runs the script in the lint-staged .

---

Ahh well in that case, I don't think there's a way around it since it's a git feature and we're in a monorepo, so it's all or nothing. But if they can be conditionally run, what's the big deal?

---

I think let's leave it out for now, we should have the Github checks working soon anyway.

---

Yeah tbf probably more important/better to have CI/CD blocking checks on PR to make sure the tests won't get skipped anyway.